edge-modules/azure-monitor/src/MetricsScrapeAndUpload.cs

```diff
@@ -25,7 +25,7 @@ public MetricsScrapeAndUpload(MetricsScraper scraper, IMetricsPublisher publishe
 
         public void Start(TimeSpan scrapeAndUploadInterval)
         {
-            this.periodicScrapeAndUpload = new PeriodicTask(this.ScrapeAndUploadMetricsAsync, scrapeAndUploadInterval, new TimeSpan(0, 0, 0), LoggerUtil.Writer, "Scrape and Upload Metrics", true);
+            this.periodicScrapeAndUpload = new PeriodicTask(this.ScrapeAndUploadMetricsAsync, scrapeAndUploadInterval, TimeSpan.FromMinutes(1), LoggerUtil.Writer, "Scrape and Upload Metrics", true);
```

Comment:
Why did this change from immediate to one minute delay? Is that part of the fix?

Reply:
Yep part of the fix. If messages are sent too quickly, the first send will hang. I couldn't figure out why, and since the SDK behavior can be weird I chose to just wait.